### PR TITLE
Add bzl file to control CXX11_ABI

### PR DESCRIPTION
**Pull request by kyleabeauchamp**
So for my miniconda3 builds to link properly, I find that I constantly have to flip `D_GLIBCXX_USE_CXX11_ABI=1` in the custom_ops files.  I'd love some way to make this easier.  

Here is one possible approach, which involves factoring out this flag into a variable stored in a `.bzl` file.  

I'm open to better approaches, but please provide explicit links to SO posts or docs that explain the usage, as I've found the bazel docs to be a bit...sparse.

### CODE REVIEW

**Comment by kyleabeauchamp** [comment]
See #174 for an example of the relevant issue.

**Comment by seanpmorgan** [comment]
I think it's a great idea, and we've briefly discussed a shared bazel wrapper library. See https://github.com/tensorflow/custom-op/issues/8

Let me think about the implementation for a few days, but I think something that is more or less what you're proposing.

**Comment by kyleabeauchamp** [comment]
Anyone else have thoughts on how to implement this?  The current code structure is kinda painful because any addons development I do requires having a branch with 3 different BUILD files edited...

**Comment by seanpmorgan** [comment]
This will be brought up in the monthly meeting this Friday 1pm PDT:
https://docs.google.com/document/d/1kxg5xIHWLY7EMdOJCdSGgaPu27a9YKpupUz2VTXqTJg/edit

I'm leaning toward a library of bazel wrappers but want to ask if we should house it internally or if there should be an external library for all custom-op repos.

**Comment by seanpmorgan** [comment]
Update from the monthly meeting: We're going to start building some bazel macros for Addons and as it matures we can split it out into its own repository. Possibly managed by SIG Build.

So for this PR could you make the file name more generic so it can hold more macros than just this one? Additionally, there is a flag `tf.sysconfig.CXX11_ABI_FLAG` so we can have this automatically configured in `configure.sh` Its up to you if you want to include that in this PR or we can do it separately. Thanks!

**Comment by kyleabeauchamp** [comment]
I've renamed the macro file as `addons_bazel_macros.bzl`

**Comment by kyleabeauchamp** [comment]
I'm trying to understand `tf.sysconfig`---isn't that set by the upstream TF build that's installed?  In my case, I'm using the pip-installed TF build but modifying the ABI flag for my addons build, which is built from source.  Thus, if my mental model is right, propagating `tf.sysconfig.CXX11_ABI_FLAG` won't help me.

**Comment by kyleabeauchamp** [comment]
I've implemented the first half of the sysconfig setup.  However, I'm not going to be able to finish it because I literally cannot find any similar usage of bazelrc in the bazel docs.  So I guess my recommendation is review as-is and finish in a follow-up PR?

**Comment by seanpmorgan** [comment]
Thanks @kyleabeauchamp I'll be able to run and test tomorrow, but I did want to ask... When do we imagine the compiled ABIs should not match TF core? Is there only a mismatch right now because TF2 preview in not on conda?

**Comment by kyleabeauchamp** [comment]
I think I mis-spoke about that particular point and have deleted the comment.  I would imagine we do want them in sync.

File: configure.sh [hunk 1]
```diff
@@ -45,7 +45,10 @@ pip install $QUIET_FLAG -r requirements.txt
 TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
 TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
 TF_SHAREDLIB=( $(python -c 'import tensorflow as tf; print(tf.sysconfig.get_link_flags()[-1])') )
+TF_CXX11_ABI_FLAG=( $(python -c 'import tensorflow as tf; print(tf.sysconfig.CXX11_ABI_FLAG)') )
 
 write_action_env_to_bazelrc "TF_HEADER_DIR" ${TF_CFLAGS:2}
 write_action_env_to_bazelrc "TF_SHARED_LIBRARY_DIR" ${TF_LFLAGS:2}
 write_action_env_to_bazelrc "TF_SHARED_LIBRARY_NAME" ${TF_SHAREDLIB:3}
+write_action_env_to_bazelrc "TF_CXX11_ABI_FLAG" ${TF_CXX11_ABI_FLAG:1}
```
Comment (seanpmorgan) [inline configure.sh]: Just `${TF_CXX11_ABI_FLAG}` as it's just one character 0 or 1

Comment (kyleabeauchamp) [reply configure.sh]: Fixed

**Pull request merged**